governance/program/src/state/proposal.rs

```diff
@@ -243,51 +248,90 @@ impl Proposal {
         }
     }
 
+    /// Calculates max vote weight for given mint supply and realm config
+    fn get_max_vote_weight(
+        &mut self,
+        realm_data: &Realm,
+        governing_token_mint_supply: u64,
+    ) -> Result<u64, ProgramError> {
+        // max vote weight fraction is only used for community mint
+        if Some(self.governing_token_mint) == realm_data.config.council_mint {
+            return Ok(governing_token_mint_supply);
+        }
+
+        match realm_data.config.community_mint_max_vote_weight_source {
+            MintMaxVoteWeightSource::SupplyFraction(fraction) => {
+                if fraction == MintMaxVoteWeightSource::SUPPLY_FRACTION_BASE {
+                    return Ok(governing_token_mint_supply);
+                }
+
+                let max_vote_weight = (governing_token_mint_supply as u128)
+                    .checked_mul(fraction as u128)
+                    .unwrap()
+                    .checked_div(MintMaxVoteWeightSource::SUPPLY_FRACTION_BASE as u128)
+                    .unwrap() as u64;
+
+                // When the fraction is used it's possible we can go over the calculated max_vote_weight
+                // and we have to adjust it in case more votes have been cast
+                let total_vote_count = self
+                    .yes_votes_count
+                    .checked_add(self.no_votes_count)
+                    .unwrap();
+
+                Ok(max_vote_weight.max(total_vote_count))
+            }
+            _ => Err(GovernanceError::VoteWeightSourceNotSupported.into()),
```

Comment:
nit: can this be `MintMaxVoteWeightSource::Absolute`?

Reply:
done